netmiko/scp_handler.py

```diff
@@ -217,16 +220,23 @@ def _remote_file_size_unix(self, remote_cmd="", remote_file=None):
 
         self.ssh_ctl_chan._enter_shell()
         remote_out = self.ssh_ctl_chan.send_command(remote_cmd, expect_string=r"[\$#]")
+        self.ssh_ctl_chan._return_cli()
+
+        if "No such file or directory" in remote_out:
+            raise IOError("Unable to find file on remote system")
+
         escape_file_name = re.escape(remote_file)
         pattern = r"^.* ({}).*$".format(escape_file_name)
         match = re.search(pattern, remote_out, flags=re.M)
         if match:
             # Format: -rw-r--r--  1 pyclass  wheel  12 Nov  5 19:07 /var/tmp/test3.txt
             line = match.group(0)
             file_size = line.split()[4]
+            return int(file_size)
 
-        self.ssh_ctl_chan._return_cli()
-        return int(file_size)
+        raise ValueError(
+            "Search pattern not found for remote file size during SCP transfer."
```

Comment:
My Suspect for SCP hang issue is that here pattern is not found and its hung there